### PR TITLE
fix: Escape strings in federated graph SDL

### PR DESCRIPTION
**Pull request by Finistere**
# Description

Please include a summary of the change and which issue is fixed.
Please also include relevant motivation and context.

# Type of change

- [ ] 💔 Breaking
- [ ] 🚀 Feature
- [ ] 🐛 Fix
- [ ] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [ ] 📖 Requires documentation update

### CODE REVIEW

**Review by fbjork — APPROVED** [review]

File: engine/crates/federated-graph/src/render_sdl.rs [hunk 1]
```diff
@@ -585,36 +583,103 @@ impl Display for Description<'_> {
     }
 }
 
+// Copy-pasted from async-graphql-value
+fn write_quoted(sdl: &mut impl Write, s: &str) -> fmt::Result {
+    sdl.write_char('"')?;
+    for c in s.chars() {
+        match c {
+            '\r' => sdl.write_str("\\r"),
```
Comment (jakubadamw) [inline engine/crates/federated-graph/src/render_sdl.rs:591]:
```suggestion
            c @ '\r' | '\n' | '\t' | '"' | '\\' => {
                sdl.write_char('\\');
                sdl.write_char(c);
            }
```

**Review by jakubadamw — APPROVED** [review]
A small simplification. Looks great otherwise!

File: engine/crates/federated-graph/src/render_sdl.rs (hunk 1 above)
Comment (jakubadamw) [reply engine/crates/federated-graph/src/render_sdl.rs:591]: And without the few further match arms. 🙂

Comment (Finistere) [reply engine/crates/federated-graph/src/render_sdl.rs:591]: indeed good catch

**Pull request merged**